shared/provision/forgot-username/index.tsx

```diff
@@ -1,93 +1,142 @@
-// TODO remove Container
 import * as React from 'react'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
-import Container from '../../login/forms/container'
-import * as Constants from '../../constants/provision'
+import PhoneInput from '../../signup/phone-number/phone-input'
+import {SignupScreen, errorBanner} from '../../signup/common'
 
 type Props = {
   forgotUsernameResult: string
   onBack: () => void
-  onSubmit: (email: string) => void
+  onSubmit: (email: string, phone: string) => void
+  waiting: boolean
 }
 
-type State = {
-  email: string
-}
+const ForgotUsername = (props: Props) => {
+  const [emailSelected, _setEmailSelected] = React.useState(true)
+  const [phoneSelected, _setPhoneSelected] = React.useState(false)
+  const setEmailSelected = React.useCallback(
+    (newState: boolean) => {
+      _setEmailSelected(newState)
+      _setPhoneSelected(!newState)
+    },
+    [_setEmailSelected, _setPhoneSelected]
+  )
+  const setPhoneSelected = React.useCallback(
+    (newState: boolean) => {
+      _setEmailSelected(!newState)
+      _setPhoneSelected(newState)
+    },
+    [_setEmailSelected, _setPhoneSelected]
+  )
 
-class ForgotUsername extends React.Component<Props, State> {
-  state = {email: ''}
+  const [email, setEmail] = React.useState('')
+  const [phoneNumber, setPhoneNumber] = React.useState<string | null>(null)
```

Comment:
why null and not just empty string? If someone has entered some phone number and then deleted it again, we don't want to be willing to send this.

Reply:
onChange from the phone number input returns (text, valid) in the callback. If valid is false, setPhoneNumber(null) is executed, therefore the only time when "phoneNumber" is a string, it must be a valid number.

If you think that this is too complicated I can make it an empty string instead.

Reply:
Maybe add a comment to that effect somewhere, but sounds fine to me.